pkg/controller/persistentvolume/controller_test.go

```diff
@@ -92,14 +83,7 @@ func TestControllerSync(t *testing.T) {
 			func(ctrl *PersistentVolumeController, reactor *volumeReactor, test controllerTest) error {
 				obj := ctrl.volumes.store.List()[0]
 				volume := obj.(*api.PersistentVolume)
-				// Remove the volume from list of resulting volumes.
-				delete(reactor.volumes, volume.Name)
-				// Poke the controller with deletion event. Cloned volume is
-				// needed to prevent races (and we would get a clone from etcd
-				// too).
-				clone, _ := conversion.NewCloner().DeepCopy(volume)
-				volumeClone := clone.(*api.PersistentVolume)
-				reactor.volumeSource.Delete(volumeClone)
+				reactor.deleteVolumeEvent(volume)
```

Comment:
Just want to make sure, we don't need add any lock on line 103 and 107 (in your new version), right?